df.history.xml

```diff
@@ -2234,29 +2234,30 @@
       <int32_t name='entity' ref-target='historical_entity' comment='the guild to which the figure belongs?'/>
       <int32_t name='source_site' ref-target='world_site'/>
       <int32_t name='dest_site' ref-target='world_site'/>
-      <int32_t/>
-      <int32_t/>
-      <int32_t/>
-      <int32_t comment='value of the trade?'/>
+      <int32_t name='production_zone'/>
+      <int32_t name='allotment'/>
+      <int32_t name='allotment_index'/>
+      <int32_t name='account_shift'/>
     </class-type>
 
     <class-type type-name='history_event_add_entity_site_profile_flagst' inherits-from='history_event' since='v0.47.01'>
       <int32_t name='entity' ref-target='historical_entity'/>
       <int32_t name='site' ref-target='world_site'/>
-    </class-type>
-
-    <class-type type-name='history_event_add_hf_entity_honorst' inherits-from='history_event' since='v0.47.01'>
-      <int32_t name='entity' ref-target='historical_entity'/>
-      <int32_t name='hf' ref-target='historical_figure'/>
-      <int32_t comment='index or identifier for the honor?'/>
+      <int32_t comment='always 65536'/>
```

Comment:
This is more likely to be two 16-bit integers, then (that would be `0` followed by `1`, I think). 4 8-bit integers or some combination of 8/16 is also a possibility.

Reply:
So far, any change that I've made to the value of this field (or to the values of either of the two fields when I split it into 2 16-bit fields) results in a change of the event sentence from this, for example:
```The Faith of Irons declared Brushedhandles to be a holy city.```
to this:
```In 155, The Faith of Irons.```
So it makes it difficult to say what's going on with this one. I agree that it _feels_ like it's two 16-bit fields.

Reply:
My spontaneous reaction to that is that it's probably two or more fields that together referenced something. Given that <0, 1> results in a reference to a holy city, I'd check for things like lists of holy cities or groups that declare such cities, and similar. Could Brushedhandles be holy city 0 or 1? In particular I'd change 0 to 1 and 1 to 0 (in separate tests) to see if the name comes out different. 0 -> 1 might not work if you only have a single holy city in your world (I've never seen any in the ones I've generated, although I haven't searched for them directly).

Reply:
it was `ffffd2d2` for me, which is a 16-bit `-1` followed by an uninitialized value

Reply:
Was yours a holy city, @BenLubar? -1 designating NULL would fit if it wasn't, although the uninitialized garbage isn't encouraging.

Reply:
if you were seeing `[0, 1]`, it's possible that `[-1, ?]` means the second field is never used anyway

Reply:
Yes, but it's sloppy, and most of the time Toady assigns a NULL or default value to unused fields, but it can't be ruled out.

Reply:
@PatrikLundell I did try to swap the 0 and 1 between fields and can't get a valid sentence out of the event after making the change.  @BenLubar I have never seen that particular value--or any value besides <0, 1>, but i'll run some more worlds to look for variations.

I'll spend some time looking the world and relevant entities to see if there's anything to correlate the values with.  It does make sense that the first value could be some kind of index/id and the second value is a flag being set.

Reply:
I'm probably thinking of a different structure entirely